doc/whats_new.rst

```diff
@@ -94,6 +94,11 @@ Bug fixes
      (`#7680 <https://github.com/scikit-learn/scikit-learn/pull/7680>`_).
      By `Ibraim Ganiev`_.
 
+   - Remove params changing inside of `transform` method of :class:`decomposition.NMF`
```

Comment:
The following would suffice:

```
    - Fixed a bug where :class:`decomposition.NMF` sets its `n_iters_`
      attribute in `transform()`. :issue:`7553` by `Ekaterina Krivich`_.
```

And under Enhancements, something like "check_estimator now attempts to ensure that methods `transform`, `predict`, etc. do not set attributes on the estimator."